### PR TITLE
Bump cluster-api to v1.1.0-beta.1 and update e2e test config

### PR DESCRIPTION
**Pull request by sedefsavas**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

**Release note**:

```release-note
Bumps cluster-api to v1.1.0-beta.1 
```

### CODE REVIEW

**Comment by invidian** [comment]
It seems this patch fixes e2e test, at least locally for me.
```diff
diff --git test/e2e/data/infrastructure-aws/kustomize_sources/topology/clusterclass-quick-start.yaml test/e2e/data/infrastructure-aws/kustomize_sources/topology/clusterclass-quick-start.yaml
index eded45a3..f86eaecb 100644
--- test/e2e/data/infrastructure-aws/kustomize_sources/topology/clusterclass-quick-start.yaml
+++ test/e2e/data/infrastructure-aws/kustomize_sources/topology/clusterclass-quick-start.yaml
@@ -137,16 +137,6 @@ metadata:
 spec:
   template:
     spec:
-      # The replicas value be overridden by the corresponding
-      # Cluster's spec.topology.controlPlane.replicas.
-      replicas: 1
-      machineTemplate:
-        # The infrastructureRef will be overridden by the corresponding
-        # ClusterClass's spec.controlPlane.machineInfrastructure.
-        infrastructureRef:
-          kind: AWSMachineTemplate
-          apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
-          name: quick-start-control-plane
       kubeadmConfigSpec:
         clusterConfiguration:
           apiServer:
@@ -165,9 +155,6 @@ spec:
             name: '{{ ds.meta_data.local_hostname }}'
             kubeletExtraArgs:
               cloud-provider: aws
-      # The version value will be overridden by the corresponding
-      # Cluster's spec.topology.version.
-      version: v1.21.2
 ---
 apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
 kind: AWSMachineTemplate
```

**Comment by invidian** [comment]
Another CI failure:
```
[5]     Error: unsupported management cluster server version: v1.23.0 - versions greater or equal than v1.22.0 are not supported
```

**Comment by invidian** [comment]
With https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/2271/commits/9fa90598aefb10b8b614cc54dd45b2d927dd9bdf, e2e tests pass.

**Comment by sedefsavas** [comment]
The reason it was failing during v1alpha3 upgrade test is that v1alpha3 does not support > 1.22 Kubernetes versions.
This is handled in CAPI by setting a different value in test-infra ([here](https://github.com/kubernetes/test-infra/blob/e9f391ba1562ceef64356d7d8c79e3290701843a/config/jobs/kubernetes-sigs/cluster-api/cluster-api-periodics-main.yaml#L95))

Ideally, we should make a change in ClusterctlUpgradeSpec to get init version as parameter similar to init binary.

**Comment by sedefsavas** [comment]
Thanks @invidian 
Updated ClusterClass test. Removed fields in cluster-api is in this PR:
https://github.com/kubernetes-sigs/cluster-api/commit/0f7e22dba4c1b970dfa64c76e236285362a3eb1f

File: test/e2e/data/shared/v1beta1_provider/metadata.yaml [hunk 1]
```diff
@@ -20,3 +20,9 @@ releaseSeries:
   - major: 1
     minor: 0
     contract: v1beta1
+  - major: 1
+    minor: 1
+    contract: v1beta1
+  - major: 1
+    minor: 2
+    contract: v1beta1
```
Comment (invidian) [inline test/e2e/data/shared/v1beta1_provider/metadata.yaml:28]: Missing newline.

**Pull request merged**